src/pages/experiments/[experimentId]/plots-and-tables/frequency/index.jsx

```diff
@@ -0,0 +1,208 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Row,
+  Col,
+  Space,
+  Collapse,
+  Skeleton,
+  Spin,
+  Radio,
+} from 'antd';
+import { useRouter } from 'next/router';
+import { useSelector, useDispatch } from 'react-redux';
+import { Vega } from 'react-vega';
+import DimensionsRangeEditor from '../components/DimensionsRangeEditor';
+import TitleDesign from '../components/TitleDesign';
+import AxesDesign from '../components/AxesDesign';
+import FontDesign from '../components/FontDesign';
+import LegendEditor from '../components/LegendEditor';
+import SelectData from './components/SelectData';
+import generateSpec from '../../../../../utils/plotSpecs/generateFrequencySpec';
+import Header from '../components/Header';
+import isBrowser from '../../../../../utils/environment';
+import {
+  updatePlotConfig,
+  loadPlotConfig,
+} from '../../../../../redux/actions/plots/index';
+import PlatformError from '../../../../../components/PlatformError';
+import loadCellSets from '../../../../../redux/actions/cellSets/loadCellSets';
+
+const { Panel } = Collapse;
+const plotUuid = 'frequencyPlotMain';
+const plotType = 'frequency';
+const route = {
+  path: 'frequency',
+  breadcrumbName: 'Frequency plot',
+};
+
+const frequencyPlot = () => {
+  const dispatch = useDispatch();
+  const config = useSelector((state) => state.plots[plotUuid]?.config);
+  const cellSets = useSelector((state) => state.cellSets);
+  const { loading, error } = cellSets;
+  const { hierarchy, properties } = cellSets;
+  const router = useRouter();
+  const { experimentId } = router.query;
+  const [plotSpec, setPlotSpec] = useState({});
+  const [plotData, setPlotData] = useState();
+
+  useEffect(() => {
+    if (!experimentId || !isBrowser) {
+      return;
+    }
+    dispatch(loadCellSets(experimentId));
+
+    dispatch(loadPlotConfig(experimentId, plotUuid, plotType));
+  }, [experimentId]);
+
+  useEffect(() => {
+    if (!config || cellSets.loading) {
+      return;
+    }
+    setPlotSpec(generateSpec(config));
+    setPlotData(generateData());
+  }, [config, properties]);
+
+  const generateData = () => {
+    const data = [];
+    if (!loading) {
+      let i = 0;
+      let sum = 0;
+      const chosenClusters = hierarchy.filter((cluster) => (
+        cluster.key === config.chosenClusters))[0].children;
+      for (; i < chosenClusters.length; i += 1) {
+        sum += properties[chosenClusters[i].key].cellIds.size;
+      }
```

Comment:
to make the code cleaner, extract this into a separate function called `calculateSum` and put the code there. In that function, instead of having this `for (; i < chosenClusters.length; i += 1)`, make it use `chosenClusters.forEach( ... ) `